src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -110,9 +126,20 @@ public <T> Future<Void> processEvent(T event) {
             return Future.succeededFuture();
         }
 
-        final String greenbidsId = UUID.randomUUID().toString();
         final String billingId = UUID.randomUUID().toString();
 
+        final Map<String, Ortb2ImpExtResult> analyticsResultFromAnalyticsTag = extractAnalyticsResultFromAnalyticsTag(
+                bidResponse);
```

Comment:
analyticsResultFromAnalyticsTag == null check happens at least 3 times further in the code, could you check it once, and skip everything that is redundant in that case?

Reply:
@EvgeniiMunin ^^

Reply:
https://github.com/prebid/prebid-server-java/pull/3242/files#r1773247031

Reply:
Have removed one redundant null check in `createBidMessage`. 

Now we have the one in `greenbidsId`
```java
private String greenbidsId(Map<String, Ortb2ImpExtResult> analyticsResultFromAnalyticsTag) {
        return Optional.ofNullable(analyticsResultFromAnalyticsTag) // <--- null check 1
                .map(Map::values)
                .map(Collection::stream)
                .flatMap(Stream::findFirst)
                .map(Ortb2ImpExtResult::getGreenbids)
                .map(ExplorationResult::getFingerprint)
                .orElseGet(() -> UUID.randomUUID().toString());
    }

```

And another one in createAdUnit
```java
final Ortb2ImpResult ortb2ImpResult = Optional.ofNullable(analyticsResultFromAnalyticsTag)
                .map(analyticsResult -> analyticsResult.get(imp.getId()))
                .map(Ortb2ImpResult::of)
                .orElse(null);
```

I prefer to keep 2 checks because of the following

if RTD module is called -> then we populate analyticsResultFromAnalyticsTag  and extract it in AnalyticsAdapter
- here in processEvent greenbidsId = ExplorationResult.getFingerprint
- then in createAdunit we populate Ortb2ImpResult  with ExplorationResult and tid for each adunit

if RTD module is not called but AnalyticsAdapter is enabled -> then in AnalyticsAdapter analyticsResultFromAnalyticsTag = null  but it does not mean we skip it. We should still populate the fields as follows
- greenbidsId is random = UUID.randomUUID()
- and Ortb2ImpResult = null for each adunit

should still keep the check for greenbidsId and another one for Ortb2ImpResult extracted from analyticsResultFromAnalyticsTag